text/0018-simplify-installer.md

```diff
@@ -0,0 +1,172 @@
+* Docs: Remove the experimental tutorial about reusing an existing ingress-gateway (e.g., istio)
+* Docs/Tutorials: Update docs for installation with a troubleshooting guide if API can not be reached (e.g., refer to Kubernetes services and description of ClusterIP vs. NodePort vs. LoadBalancer) 
+* Docs/Tutorials: Provide solid Docs for Demo/POC (e.g., with GKE, K3s) and setting up a full installation of Keptn with onboarding services 
+
+## Trade-offs and mitigations
+
+### Each Kubernetes instance or Cloud-provider has different LoadBalancers (or non at all)
+
+Each platform has different ways of using the service types NodePort and LoadBalancer, but we can tackle this with recommending the preferred option for the respective platform (e.g., GKE, AKS, EKS – Loadbalancer; MiniKube, K3s, MicroK8s: NodePort; OpenShift, MiniShift: ???) 
+
+### Keptn API can not be reached after installation
+
+This is a problem that already occurs sometimes, but in this scenario it could occur much more often. However, in this case we can refer the user to the documentation of Kubernetes services and types, and how to use them to expose a service.
+
+### Change the behaviour afterwards
+
+If the user wants to change this behaviour afterwards, it is as easy as calling `kubectl –n keptn edit service api` (`kubectl –n keptn patch service api …`) and change from ClusterIP to LoadBalancer. 
+
+* Change api service type to LoadBalancer
+    ```console
+    kubectl patch svc api-gateway-nginx -n keptn -p '{"spec": {"type": "LoadBalancer"}}'
+    ```
+* Change api service type to NodePort
+    ```console
+    kubectl patch svc api-gateway-nginx -n keptn -p '{"spec": {"type": "NodePort"}}'
+    ```
+* Change bridge service type to LoadBalancer
+    ```console
+    kubectl patch svc bridge -n keptn -p '{"spec": {"type": "LoadBalancer"}}'
+    ```
+* Change bridge service type to NodePort
+    ```console
+    kubectl patch svc bridge -n keptn -p '{"spec": {"type": "NodePort"}}'
+    ```
+
+If the user wants to use an ingress-controller (e.g., nginx-ingress, Linkerd, istio, traefic) for API/Bridge, the recommended way is to use ClusterIP and apply an ingress manifest manually. 
+
+
+## Breaking changes
+
+* The CLI command `keptn configure bridge` will need to be simplified to no longer change any service types or ingress manifests.
+* The configmap for `KEPTN_DOMAIN` will no longer be available, as we don't know the domain when installing Keptn.
+* Helm-Service will break without `KEPTN_DOMAIN` - instead we should ask the user to supply this configuration using a new environment variable called `INGRESS_HOSTNAME_SUFFIX` which we will default to `svc.cluster.local`
+* Helm-Service will need to check if istio is available before applying any virtualservices
+* Not sure, but OpenShift support might break and we need to fix certain things
+* Dynatrace-service will break as it also relies on `KEPTN_DOMAIN` - especially for problem notifications
```

Comment:
In addition to Problem notifications, the `KEPTN_COMAIN` is currently used during the dashboard setup (to provide links to the onboarded services, and to the bridge), as well as to provide deep links to the bridge in dynatrace events.

For the first two cases, we could pass the URL of the API/Bridge as parameter(s) for the `configure monitoring` command.
For adding the deep links to the bridge in deployment events, we will have to rely on the `INGRESS_HOSTNAME_SUFFIX` env var (as we also do in the helm-service)

Reply:
Thanks for bringing this up, I've added it to the KEP.
Basically, with the proposed change in this KEP deep links to Keptn Bridge will not work anymore, as Keptn does not expose Bridge via a VirtualService or alike.